bin/requirements.rb

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env ruby
+
+# This script takes the existing requirements.txt file
+# And updates them for our supported packages
+#
+# USAGE:
+# 1. Get all module requirements (don't include documentation or testing ones)
+#
+#     ./requirements.rb ./requirements.txt /usr/lib/python3.9/site-packages/ansible_collections/ > new_requirements.txt
+#
+# 2. Resolve conflicts and determine if new one is correct
+#
+#     diff {,new-}requirements.txt
+#     # cp new_requirements.txt requirements.txt
+#
+
+class ParseRequirements
+  PACKAGES=%w[
+    amazon/aws/requirements.txt
+    ansible/netcommon/requirements.txt
+    ansible/utils/requirements.txt
+    awx/awx/requirements.txt
+    azure/azcollection/requirements-azure.txt
+    cisco/intersight/requirements.txt
+    community/aws/requirements.txt
+    community/okd/requirements.txt
+    community/vmware/requirements.txt
+    google/cloud/requirements.txt
+    kubernetes/core/requirements.txt
+    openstack/cloud/requirements.txt
+    ovirt/ovirt/requirements.txt
+    theforeman/foreman/requirements.txt
+    vmware/vmware_rest/requirements.txt
+  ]
+  attr_reader :filenames, :non_modules, :final
+  def initialize
+    @filenames = []
+    @non_modules = []
+
+    @final = {}
+  end
+
+  def add_target(filename_or_dirname)
+    if Dir.exist?(filename_or_dirname)
+      add_file(filename_or_dirname)
+    else
+      add_dir(filename_or_dirname)
+    end
+  end
+
+  def add_file(filename)
+    @filenames << filename
+    @non_modules << filename
+
+    self
+  end
+
+  def add_dir(dirname)
+    dirname = dirname[0..-2] if dirname.end_with?("/")
+    PACKAGES.each do |package|
+      filename = "#{dirname}/#{package}"
+      if File.exist?(filename)
+        @filenames << filename
+      else
+        $stderr.puts("NOTICE: missing #{filename}")
+      end
+    end
+
+    self
+  end
+
+  def parse
+    filenames.each do |fn|
+      # the list of requirements-files can have items commented out - ignore those
+
+      mod = module_name_from_filename(fn)
+      IO.read(fn).split("\n").each do |line|
```

Comment:
Doesnt really matter but:

```suggestion
      File.foreach(fn, chomp: true) do |line|
```

This may help if we encounter any windows line endings in the future.